Data/models/Bookmark.swift

```diff
@@ -171,6 +182,20 @@ public final class Bookmark: NSManagedObject, WebsitePresentable, Syncable, CRUD
         }
     }
     
+    /// WARNING: This method deletes all current favorites and replaces them with new one from the array.
+    public class func replaceFavorites(with favorites: [(url: URL, title: String)]) {
```

Comment:
Maybe adjust this name a bit so it has stronger connotation: `forceOverwriteFavorites` or something?